src/RegExLib.Core/Entities/Author.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections.Generic;
+using RegExLib.Core.Events;
+using RegExLib.Core.Interfaces;
+using RegExLib.SharedKernel;
+
+namespace RegExLib.Core.Entities
+{
+    public class Author : BaseEntity
+    {
+        public string FullName { get; set; }
+        public string Username { get; set; }
```

Comment:
@antmdvs
Thanks for your review :)
For this, I replaced "@" when setting UserName in the constructor or there is another better way than that.

```csharp
       public Author(string userId, string username, string fullName)
        {
            UserId = userId;
            Username = username.Replace("@", "");
            FullName = fullName;
        }
```

Reply:
Happy to help :) I was assuming @ardalis intended to use [Domain-Driven Design](https://martinfowler.com/bliki/DomainDrivenDesign.html) for this since his clean architecture solution supports it with concepts like Entities & ValueObjects. That is, despite the domain being relatively simple (I think DDD is usually most appropriate for complex domains and Steve might think it overkill for this....?).

That said, I was asking from a DDD perspective if instead of using a primitive string type for `Username`, whether it should be a `public class UserName : ValueObject` type (@ardalis ?). That way, the validation could be encapsulated in the `UserName` value object itself. FYI, I'm trying to practice DDD, so that's why I'm asking Steve these questions in your PR.

DDD or not, I feel like the approach here might really be a "business" decision. Should @ be silently stripped as you've done above or should the user be alerted that they have requested an invalid user name and allowed to change it?

Reply:
...Now to clarify my initial comment, I was assuming @ardalis intended to use [Domain-Driven Design](https://martinfowler.com/bliki/DomainDrivenDesign.html) for this since his clean architecture solution supports it with concepts like Entities & ValueObjects. That is, despite the domain being relatively simple (I think DDD is usually most appropriate for complex domains and Steve might think it overkill for this....?).

That said, I was asking from a DDD perspective if instead of using a primitive string type for `Username`, whether it should be a `public class UserName : ValueObject` type (@ardalis ?). That way, the validation could be encapsulated in the `UserName` value object itself. FYI, I'm trying to practice DDD, so that's why I'm asking Steve these questions in your PR.

Reply:
I am not sure if we use ValueObject for UserName, EF will work with that or not.
I checked also DevBetterWeb how they create the Entities I see they did not use ValueObject.

Reply:
Deferring to @ardalis on this, as I may have a misunderstanding of how validation works with VO, nor am I familiar with the EF aspect.

Reply:
https://github.com/ardalis/RegExLib.com/pull/14#discussion_r447193020